src/copilot/contextProvider.ts

```diff
@@ -0,0 +1,211 @@
+            "extension": getExtensionName(),
+            "status": "succeeded",
+            "installCount": installCount
+        });
+    }
+    catch (error) {
+        logger.error('Error occurred while registering Java context provider for GitHub Copilot extension:', error);
+    }
+}
+
+async function resolveJavaContext(_request: ResolveRequest, _token: vscode.CancellationToken): Promise<SupportedContextItem[]> {
+    const items: SupportedContextItem[] = [];
+    const start = performance.now();
+    try {
+        // Get current document and position information
+        const activeEditor = vscode.window.activeTextEditor;
+        if (!activeEditor || activeEditor.document.languageId !== 'java') {
+            return items;
+        }
+
+        const document = activeEditor.document;
+
+        // 1. Project basic information (High importance)
+        const javaVersion = await getProjectJavaVersion(document);
+
+        items.push({
+            name: 'java.version',
+            value: javaVersion,
+            importance: 90,
+            id: 'java-version',
+            origin: 'request'
+        });
+
+        items.push({
+            name: 'java.file',
+            value: vscode.workspace.asRelativePath(document.uri),
+            importance: 80,
+            id: 'java-file-path',
+            origin: 'request'
+        });
+
+        // Try to get cached imports first
+        let importClass = await contextCache.get(document.uri);
+        if (!importClass) {
+            // If not cached, resolve and cache the result
+            const resolvedImports = await CopilotHelper.resolveLocalImports(document.uri);
+            logger.trace('Resolved imports count:', resolvedImports);
+            if (resolvedImports) {
+                // Get current caret offset for position-sensitive caching
+                const currentCaretOffset = vscode.window.activeTextEditor?.document.offsetAt(
+                    vscode.window.activeTextEditor.selection.active
+                );
+                
+                await contextCache.set(document.uri, resolvedImports, undefined, currentCaretOffset);
+                importClass = resolvedImports;
+                logger.trace('Cached new imports, cache size:', importClass.length);
+            }
+        } else {
+            logger.info('Using cached imports in resolveJavaContext, cache size:', importClass.length);
+        }
+        
+        if (importClass) {
+            for (const cls of importClass) {
+                items.push({
+                    uri: cls.uri,
+                    value: cls.className,
+                    importance: 70,
+                    origin: 'request'
+                });
+            }
+        }
+    } catch (error) {
+        logger.error('Error resolving Java context:', error);
+    }
+    logger.info('Total context resolution time:', performance.now() - start, 'ms', ' ,size:', items.length);
+    logger.info('Context items:', items);
```

Comment:
I assume the context provider will be called very frequently, this will print too many verbose logs. If it's the debug purpose, you should only print it when debugging extension host.